onnxruntime/hosting/executor.h

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef ONNXRUNTIME_HOSTING_EXECUTOR_H
+#define ONNXRUNTIME_HOSTING_EXECUTOR_H
+
+#include <google/protobuf/stubs/status.h>
+
+#include "environment.h"
+#include "predict.pb.h"
+
+namespace onnxruntime {
+namespace hosting {
+
+class Executor {
```

Comment:
Maybe make a pure abstract base class called `Executor` and then have a `RequestExecutor` implementation. We may end up having different "executors" like batch, async, etc.

Reply:
We may not need the layer of abstraction, and different "executor" may have different signature. Actually they could be put into this class directly.

Reply:
I think the `Executor` is already an abstraction. Additional layer could be added when we refactor the code for more complex situation.